model/host/host_intent.go

```diff
@@ -0,0 +1,135 @@
+package host
+
+import (
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/distro"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/pkg/errors"
+)
+
+// HostOptions is a struct of options that are commonly passed around when creating a
+// new cloud host.
+type HostOptions struct {
```

Comment:
it's weird in go to have package names (e.g. host) appear in the name of a type (e.g. HostOptions) so maybe we should try something like "CreationOptions" for this type.